src/components/game-room/boardMap/GameBoard.tsx

```diff
@@ -44,11 +46,19 @@ const ANIMATION_CONFIG = {
 };
 
 const GameBoard = () => {
-  const { scores } = useScoreStore();
-  const { targetUser, triggerUser, eventType } = useGameBubbleStore();
-  const { participantInfo } = useParticipantInfoStore();
-  const { setSoundEvent } = useSoundEventStore();
-  const { isActiveDice } = useGameDiceStore();
+  const scores = useScoreStore(state => state.scores);
+  const { targetUser, triggerUser, eventType } = useGameBubbleStore(
+    useShallow(state => ({
+      targetUser: state.targetUser,
+      triggerUser: state.triggerUser,
+      eventType: state.eventType,
+    })),
+  );
```

Comment:
질문이 있어 남겨둡니다. `targetUser, triggerUser, eventType`의 경우 객체상태로 관리되는게 아니라, 구조분해 할당으로 인해 각각 관리되는데 이 경우에 

```suggestion
  const targetUser = useGameBubbleStore(state => state.targetUser)
  const triggerUser = useGameBubbleStore(state => state.triggerUser)
  const eventType = useGameBubbleStore(state => state.eventType)
```

과 같이 작성하는것과 다른 효과가 있을까요 ??

Reply:
세 가지 상태가 항상 함께 변경되어, 상태 간의 연관성이 높다고 판단해 shallow 방식을 사용했습니다.

`shallow 방식`과 `각각의 selector 방식`은 거의 똑같은데 아래와 같은 차이점이 있습니다!

- `shallow 방식`: 연관된 값들을 함께 구독하기 좋다. 하지만 상태 변경 시마다 얕은 비교를 수행하기 때문에 약간의 연산 비용이 발생한다.

- `각각의 selector 방식`: 객체 참조 비교만 수행하기 때문에 연산 비용이 미세하게 더 적다.

`각각의 selector 방식`을 사용한 예시로는 저희 프로젝트의 `YoutubePlayer.tsx`가 있습니다.
두 개의 songUrl을 같은 store에서 가져오지만 각각 selector 방식으로 사용하고 있습니다.

```js
// src/components/game-room/gameScreen/YoutubePlayer.tsx

const url = useGameRoundInfoStore(state => state.roundInfo.songUrl);
const url2 = useGameRoundInfoStore(state => state.roundInfo.songUrl2);
```
이처럼 속도가 매우 중요한 컴포넌트에서는 상태 연산이 성능에 영향을 줄 수 있기 때문에 `selector 방식`을 사용하고 그렇지 않은 경우에는 `shallow 방식`이 적절하다고 생각합니다.